azurerm/internal/services/sentinel/sentinel_alert_rule_fusion_resource.go

```diff
@@ -106,28 +106,35 @@ func resourceSentinelAlertRuleFusionCreateUpdate(d *schema.ResourceData, meta in
 	if !d.IsNewResource() {
 		resp, err := client.Get(ctx, workspaceID.ResourceGroup, "Microsoft.OperationalInsights", workspaceID.WorkspaceName, name)
 		if err != nil {
-			return fmt.Errorf("retrieving Sentinel Alert Rule Fusion %q (Resource Group %q / Workspace: %q): %+v", name, workspaceID.ResourceGroup, workspaceID.WorkspaceName, err)
+			return fmt.Errorf("retrieving Sentinel Alert Rule Fusion %q (Workspace: %q): %+v", name, workspaceID, err)
 		}
 
 		if err := assertAlertRuleKind(resp.Value, securityinsight.Fusion); err != nil {
-			return fmt.Errorf("asserting alert rule of %q (Resource Group %q / Workspace: %q): %+v", name, workspaceID.ResourceGroup, workspaceID.WorkspaceName, err)
+			return fmt.Errorf("asserting alert rule of %q (Workspace: %q): %+v", name, workspaceID, err)
 		}
 		params.Etag = resp.Value.(securityinsight.FusionAlertRule).Etag
 	}
 
 	if _, err := client.CreateOrUpdate(ctx, workspaceID.ResourceGroup, "Microsoft.OperationalInsights", workspaceID.WorkspaceName, name, params); err != nil {
-		return fmt.Errorf("creating Sentinel Alert Rule Fusion %q (Resource Group %q / Workspace: %q): %+v", name, workspaceID.ResourceGroup, workspaceID.WorkspaceName, err)
+		return fmt.Errorf("creating Sentinel Alert Rule Fusion %q (Workspace: %q): %+v", name, workspaceID, err)
 	}
 
 	resp, err := client.Get(ctx, workspaceID.ResourceGroup, "Microsoft.OperationalInsights", workspaceID.WorkspaceName, name)
 	if err != nil {
-		return fmt.Errorf("retrieving Sentinel Alert Rule Fusion %q (Resource Group %q / Workspace: %q): %+v", name, workspaceID.ResourceGroup, workspaceID.WorkspaceName, err)
+		return fmt.Errorf("retrieving Sentinel Alert Rule Fusion %q (Workspace: %q): %+v", name, workspaceID, err)
 	}
+
 	id := alertRuleID(resp.Value)
 	if id == nil || *id == "" {
-		return fmt.Errorf("empty or nil ID returned for Sentinel Alert Rule Fusion %q (Resource Group %q / Workspace: %q) ID", name, workspaceID.ResourceGroup, workspaceID.WorkspaceName)
+		return fmt.Errorf("empty or nil ID returned for Sentinel Alert Rule Fusion %q (Workspace: %q) ID", name, workspaceID)
+	}
+
+	resId, err := parse.AlertRuleID(*id)
```

Comment:
same here, we should be able to remove the line 126/127 down to here in favour of building this up using the  NewAlertRuleID func instead?